lib/src/interceptor.dart

```diff
@@ -29,13 +29,20 @@ class Registry {
     interceptor._onReply?.close();
   }
 
+
   Interceptor? match(HttpClientRequest request) {
     for (var interceptor in _interceptors) {
       if (interceptor._matcher.match(request as MockHttpClientRequest)) {
         return interceptor;
       }
     }
 
+    for (var autoInterceptor in _interceptors.map ((i)=> i.autoOption).where ((i)=> i!= null) ) {
```

Comment:
It would be be better not to have to create a new collection of interceptors (to provide a function such as `Matcher.defaultMatch()`) but that doesn't allow us to define a diffferent reply.  (I suppose we could delay creating the new interceptor until after the `defaultMatch()` but then what would be the matcher in the new interceptor?)

Really needs a refactor to express this properly.  I have some ideas... :-)